### ISSUE TITLE
The specified network adapter cannot be installed on your Mac.

### ISSUE THREAD

**Issue by wizonesolutions** [Parallels/vagrant-parallels#322]
I'm running into a new error in Vagrant 2.1.0:

```
There was an error while command execution. The command and stderr is shown below.

Command: ["/usr/local/bin/prlsrvctl", "net", "add", "vagrant-vnet14", "--type", "host-only"]

Stderr: Failed to add Virtual Network vagrant-vnet14: Unable to install the network adapter. The specified network adapter cannot be installed on your Mac.
```

vagrant-parallels version: 1.7.8

I'm not sure how I would try to reproduce this via the GUI. I've submitted my debug logs anyway. ID: 255283886

**Comment by legal90** [comment]
@wizonesolutions The error comes from Parallels Desktop and most likely you run out of max allowed virtual interfaces, which seems to be locked at 16.

I've managed to reproduce it manually:
```
$ prlsrvctl net add vagrant-vnet1 --type host-only
$ prlsrvctl net add vagrant-vnet2 --type host-only
$ prlsrvctl net add vagrant-vnet3 --type host-only
$ prlsrvctl net add vagrant-vnet4 --type host-only
$ prlsrvctl net add vagrant-vnet5 --type host-only
$ prlsrvctl net add vagrant-vnet6 --type host-only
$ prlsrvctl net add vagrant-vnet7 --type host-only
$ prlsrvctl net add vagrant-vnet8 --type host-only
$ prlsrvctl net add vagrant-vnet9 --type host-only
$ prlsrvctl net add vagrant-vnet10 --type host-only
$ prlsrvctl net add vagrant-vnet11 --type host-only
$ prlsrvctl net add vagrant-vnet12 --type host-only
$ prlsrvctl net add vagrant-vnet13 --type host-only
$ prlsrvctl net add vagrant-vnet14 --type host-only
$ prlsrvctl net add vagrant-vnet15 --type host-only
Failed to add Virtual Network vagrant-vnet15: Unable to install the network adapter. The specified network adapter cannot be installed on your Mac.
```

```
$ /usr/local/bin/prlsrvctl net list
Network ID        Type      Bound To
Shared            shared  vnic0
NAT server:

Host-Only         host-only  vnic1
vagrant-vnet1     host-only  vnic2
vagrant-vnet2     host-only  vnic3
vagrant-vnet3     host-only  vnic4
vagrant-vnet4     host-only  vnic5
vagrant-vnet5     host-only  vnic6
vagrant-vnet6     host-only  vnic7
vagrant-vnet7     host-only  vnic8
vagrant-vnet8     host-only  vnic9
vagrant-vnet9     host-only  vnic10
vagrant-vnet10    host-only  vnic11
vagrant-vnet11    host-only  vnic12
vagrant-vnet12    host-only  vnic13
vagrant-vnet13    host-only  vnic14
vagrant-vnet14    host-only  vnic15
<...>
```
Do you really have 15 VMs using different host-only interfaces?

**Comment by wizonesolutions** [comment]
@legal90 Yeah. They are not all active, but I guess it does not remove the old interfaces on suspend? Although I think most of them are halted. Let me try a different IP on an existing IP range, then...

**Comment by wizonesolutions** [comment]
That worked! It would be great if the plugin could detect this situation and output a clearer error message, at least—or perhaps manage networks a bit more aggressively. Because it can always recreate them, right? Especially if the VM in question is halted. If suspended, I understand wanting to keep them up, since removing them might mess up NFS?

Thanks for the quick reply!

**Comment by legal90** [comment]
Virtual interfaces are deleted only on VM deletion (`vagrant destroy`) and only if there are no other VM using the same interface.

FYI: our provider tries to optimize the number of `vnics`, so it should use the same existing vnic for multiple VMs if they are configured with `private_network`s sharing the same IP range.
For example, such 2 VMs will share the same vnic:
```
    config.vm.network "private_network", ip: "10.100.1.15", netmask: '16'

    config.vm.network "private_network", ip: "10.100.20.30", netmask: '16'
```

**Comment by AdrianwithaW** [comment]
Have just encountered this problem… what am I looking at? I only have one VM set up on my machine, but the list output by the command above is like so:
```Network ID        Type      Bound To
Shared            shared
NAT server:

Host-Only         host-only
vagrant-vnet0     host-only
vagrant-vnet1     host-only
vagrant-vnet2     host-only
vagrant-vnet3     host-only
vagrant-vnet4     host-only
vagrant-vnet5     host-only
vagrant-vnet6     host-only
vagrant-vnet7     host-only
vagrant-vnet8     host-only
vagrant-vnet9     host-only
vagrant-vnet10    host-only
vagrant-vnet11    host-only
vagrant-vnet12    host-only
vagrant-vnet13    host-only
Wi-Fi (en0)       bridged   en0
USB 10/100/1000~  bridged   en9
Default           bridged   FF:FF:FF:FF:FF:FF```

Are these instances that were halted previously? How do I delete them - assuming they're of no use?

**Comment by AdrianwithaW** [comment]
Ok, somewhat simple fix here - this list is also available in Parallels in the network preferences. They can just be deleted one by one which seems to cause no harm.

**Comment by legal90** [comment]
Hi @AdrianwithaW,
The issue you have most likely the same as #369. It was fixed it varagrant-parallels v2.1.0:
https://github.com/Parallels/vagrant-parallels/releases/tag/v2.1.0
Please make sure that you use the latest version (`vagrant plugin list`)
